### PR TITLE
feat: MLIBZ-2610 Increase code coverage of store source files

### PR DESCRIPTION
**Pull request by yuliya-guseva**
#### Description
Increase code coverage of store source files

#### Changes
Added tests for DataStore, LinkedDataStore, FilesTore, UserStore. 
Tests for deprecated mwthods weren't added.
Filed bugs: `MLIBZ-2653`, `MLIBZ-2654`

#### Tests
Instrumented

### CODE REVIEW

File: android-lib/src/androidTest/java/com/kinvey/androidTest/store/UserStoreTest.java [hunk 1]
```diff
@@ -1310,16 +1330,272 @@ private String createRandomUserName(String testName) {
...
+        assertEquals("currentUser must not be null", callback.error.getMessage());
+    }
+
+    private DefaultKinveyClientCallback convenience(final Client client) throws InterruptedException {
+        final CountDownLatch latch = new CountDownLatch(1);
+        final DefaultKinveyClientCallback callback = new DefaultKinveyClientCallback(latch);
+        LooperThread looperThread = new LooperThread(new Runnable() {
+            @Override
+            public void run() {
+                UserStore.convenience(client, callback);
```
Comment (vinaygahlawat) [inline android-lib/src/androidTest/java/com/kinvey/androidTest/store/UserStoreTest.java]: What does the `convenience` function on the `UserStore` for?

File: android-lib/src/main/java/com/kinvey/android/store/UserStore.java [hunk 2]
```diff
@@ -325,7 +325,7 @@ public static void signUp(@NonNull String username, @NonNull String password,
      * Sample Usage:
      * <pre>
      * {@code
-     *  UserStore.loginSalesForce("accessToken", mClient, new KinveyClientCallback<User>() {
+     *  UserStore.loginMobileIdentity("accessToken", mClient, new KinveyClientCallback<User>() {
```
[the remark this answers is not in the document]
Comment (yuliya-guseva) [reply android-lib/src/main/java/com/kinvey/android/store/UserStore.java]: The convenience function is the similar to `retrieve`, `convenience` gets user metadata from the backend and update `client.activeUser`, `retrieve` doesn't update `client.activeUser`.

File: android-lib/src/androidTest/java/com/kinvey/androidTest/store/UserStoreTest.java (hunk 1 above)
Comment (yuliya-guseva) [reply android-lib/src/androidTest/java/com/kinvey/androidTest/store/UserStoreTest.java]: The convenience function is the similar to `retrieve`, `convenience` gets user metadata from the backend and update `client.activeUser`, `retrieve` doesn't update `client.activeUser`.

File: android-lib/src/androidTest/java/com/kinvey/androidTest/store/DataStoreTest.java [hunk 3]
```diff
@@ -3374,4 +3394,91 @@ public void run() {
...
+        DefaultKinveyClientCallback saveCallback = save(store, createPerson(TEST_USERNAME));
+        assertNotNull(saveCallback.result.getId());
+        ids.add(saveCallback.result.getId());
+        saveCallback = save(store, createPerson(TEST_USERNAME_2));
+        assertNotNull(saveCallback.result.getId());
+        ids.add(saveCallback.result.getId());
+        DefaultKinveyReadCallback kinveyListCallback = find(store, ids, DEFAULT_TIMEOUT, new KinveyCachedClientCallback<KinveyReadResponse<Person>>() {
+            @Override
+            public void onSuccess(KinveyReadResponse<Person> result) {
+
```
[the remark this answers is not in the document]
Comment (yuliya-guseva) [reply android-lib/src/androidTest/java/com/kinvey/androidTest/store/DataStoreTest.java]: In some place we have added such checking, in some place haven't. So I haven't added it to this place. Do you think we need to add checking correct results from the cache callback?

[the remark this answers is not in the document]
Comment (vinaygahlawat) [reply android-lib/src/androidTest/java/com/kinvey/androidTest/store/DataStoreTest.java]: I think that in general, if we are using a CACHE store type, we should be checking the cached callback value as well. We don't have to retroactively go through all the tests and add the assertion, but when we encounter it, as in this case, we should make the change to check the callback result.

File: android-lib/src/androidTest/java/com/kinvey/androidTest/store/UserStoreTest.java (hunk 1 above)
Comment (vinaygahlawat) [reply android-lib/src/androidTest/java/com/kinvey/androidTest/store/UserStoreTest.java]: So in the case of `retrieve`, it only updates the credential store?

Comment (yuliya-guseva) [reply android-lib/src/androidTest/java/com/kinvey/androidTest/store/UserStoreTest.java]: `retrieve` only returns user data in the callback, but doesn't update the active user.

File: android-lib/src/androidTest/java/com/kinvey/androidTest/store/DataStoreTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (yuliya-guseva) [reply android-lib/src/androidTest/java/com/kinvey/androidTest/store/DataStoreTest.java]: I agree. Updated.

**Pull request merged**